packages/node_modules/@webex/plugin-meetings/src/metrics/mqa-processor.js

```diff
@@ -0,0 +1,118 @@
+import bowser from 'bowser';
+
+import {
+  STATS,
+  MQA_STATS
+} from '../constants';
+
+
+/**
+ * @description MQAProcessor handles interval data for MQA
+ * @export
+ * @class MQAProcessor
+ */
+class MQAProcessor {
+  /**
+     * @constructor
+     * @public
+     * @memberof MQAProcessor
+     */
+  constructor() {
+    this.data = {
+      videoReceive: [],
+      audioTransmit: [],
+      audioReceive: [],
+      videoTransmit: []
+    };
+    this.intervalNumber = 1;
+  }
+
+  /**
+   * @param {String} id
+   * @param {Array<WebRTCData>} interval - a slice of metrics history
+   * @returns {undefined}
+   * @memberof MQAProcessor
+   */
+  process(id, interval) {
+    let rtcCandidatePair, rtcOutVideo, rtpOutVideo; // TODO: , rtcInVideo, rtpInVideo, rtcOutAudio, rtcInAudio, rtpInAudio, rtpOutAudio; // TODO:
+    let vsTransmit;
+
+    const {
+      videoReceive, audioTransmit, audioReceive, videoTransmit
+    } = this.data;
+
+    const sumValue = interval[0]; // the head is the last interval value, webRTC spec has some values automatically summed
+
+    if (sumValue) {
+      rtcCandidatePair = sumValue.rtcCandidatePair;
+    }
+
+    switch (id) {
+      case STATS.AUDIO_SENDER_ID:
+        audioTransmit.push();
+        break;
+      case STATS.AUDIO_RECEIVER_ID:
+        audioReceive.push();
+        break;
+      case STATS.VIDEO_SENDER_ID:
+        videoTransmit.push();
+        break;
+      case STATS.VIDEO_RECEIVER_ID:
+        videoReceive.push();
+        break;
+      case STATS.SHARE_SENDER_ID:
+        if (sumValue) {
+          rtcOutVideo = sumValue.rtcOutVideo;
+          rtpOutVideo = sumValue.rtpOutVideo;
+        }
+        vsTransmit = {...MQA_STATS.DEFAULT_SHARE_SENDER_STATS};
```

Comment:
does `vsTransmit` need to be declared outside the switch?

Reply:
eslint :(

Reply:
You can wrap the case into a block to be able to `const vsTransmit`